dffml/util/cli/cmd.py

```diff
@@ -87,21 +89,46 @@ def add_subs(self, add_from: "CMD"):
                 except argparse.ArgumentError as error:
                     raise Exception(repr(add_from)) from error
 
+        position_list = {}
+        for field in dataclasses.fields(add_from.CONFIG):
+            arg = mkarg(field)
+            if isinstance(arg, Arg):
+                if isinstance(field.metadata.get("position"), int):
+                    position_list[field.metadata.get("position")] = (
+                        field.name,
+                        arg,
+                    )
+                else:
+                    try:
+                        if position_list:
+                            for position in sorted(position_list.keys()):
```

Comment:
I think you should do the for loop through position_list  after line 112, I suspect this only works if there are two positional arguments that don't have a non-positional one in between them.